### DISCUSSION TITLE
How to re-add a osd to ceph cluster with rook?

### DISCUSSION THREAD

**Discussion by zhucan** [rook/rook#10223] [Q&A] [answered]
1. remove device from ceph cluster
2. add the device to the ceph cluster

@travisn

**Comment by travisn** [comment, marked as answer]
@zhucan Did you see the [OSD management](https://rook.github.io/docs/rook/v1.9/ceph-osd-mgmt.html) topic?

**Reply by zhucan** [reply]
@travisn yes. Firstly: delete the osd deployment; Second: kubectl create -f purge-osd.yaml, is it correctly?

**Reply by travisn** [reply]
Yes, basically:
1. Scale down the osd deployment (if the osd pod isn't already crashing)
2. purge-osd.yaml
3. Wipe/replace the disk
4. Restart the operator to scan for new devices and create a new OSD

**Comment by zhucan** [comment]
We should wait the data rebalance after scale down the osd depoyment, maybe we need to update weigth for the osd.

**Reply by travisn** [reply]
If the OSD is already dead, there is no need to wait for rebalance. If the OSD was healthy and you want to replace it, you can first mark it `out` from the toolbox (something like `ceph osd out <osd-id>`), then wait for the rebalance before removing the OSD and replacing it.

**Reply by zhucan** [reply]
Thanks for your responses.
